ibacm/prov/acmp/src/acmp.c

```diff
@@ -2412,8 +2412,43 @@ static int acmp_add_addr(const struct acm_address *addr, void *ep_context,
 static void acmp_remove_addr(void *addr_context)
 {
 	struct acmp_addr *address = addr_context;
+	struct acmp_device *dev;
+	struct acmp_dest *dest;
+	struct acmp_ep *ep;
+	int i;
 
 	acm_log(2, "\n");
+
+	/*
+	 * The address may be a local destination address. If so,
+	 * delete it from the cache.
+	 */
+
+	pthread_mutex_lock(&acmp_dev_lock);
+	list_for_each(&acmp_dev_list, dev, entry) {
+		pthread_mutex_unlock(&acmp_dev_lock);
```

Comment:
By releasing the lock, we don't know if the device is removed immediately afterwards.  Can't we just hold the device and port locks throughout the loops?

Reply:
@shefty I'm glad you remarked on this.. I thought it was wrong too, but there were other existing examples of this pattern.. From what I could tell that acmp_dev_list is only added to (!?!) Very confusing.

Reply:
The construct was copied from another place in the same file. But the code seems to broken when hot-unplug gets supported.

As of now, my patch seems as solid as the same construct in acmp_retry_handler().